contrib/enfuse_pro_2.lua

```diff
@@ -0,0 +1,1026 @@
+	if (cmbx_output_format.value == "TIFF") then
+		cmd_suffix_output_format="tif"
+		blend_args = blend_args.." --compression="..dt.preferences.read("module_enfuse_pro", "compression_tiff", "enum")
+	elseif (cmbx_output_format.value == "JPEG") then
+		cmd_suffix_output_format="jpg"
+		blend_args = blend_args.." --compression="..truncate(dt.preferences.read("module_enfuse_pro", "compression_jpeg", "integer"))
+	elseif (cmbx_output_format.value == "PNG") then
+		cmd_suffix_output_format="png"
+	elseif (cmbx_output_format.value == "PNM") then
+		cmd_suffix_output_format="pnm"     
+	elseif (cmbx_output_format.value == "PBM") then
+		cmd_suffix_output_format="pbm"   
+	elseif (cmbx_output_format.value == "PPM") then
+		cmd_suffix_output_format="ppm"
+	end
+
+	--Set output path and add filename
+	if (chkbtn_source_location.value) then
+		cmd_output_path = source_path
+	else
+	cmd_output_path=file_chooser_button_path.value 
+	end
+	path_with_filename = cmd_output_path..os_path_seperator..first_filename.."-"..last_id.."."..cmd_suffix_output_format
+	
+	--Create unique name with index if GUI selection create unique name (don't overwrite), also do this if user selected to make variants from presets
+	if (cmbx_existing_file.selected == 1) or (from_preset) then
+		path_with_filename = df.create_unique_filename(path_with_filename)
+	end
+	
+	if dt.configuration.running_os == "windows" then
+		cmd_output_image = " --output="..path_with_filename
+	else
+		cmd_output_image = " --output="..'"'..path_with_filename..'"'
+		
+	end
+	blend_args = blend_args..cmd_output_image
+
+	images_to_blend = ""
+	for j=0, counted_images-1 do
+		if j < 10 then
+			id = "000"..tostring(j)
+		else
+			id = "00"..tostring(j)
+		end
+		if dt.configuration.running_os == "windows" then
+			images_to_blend = images_to_blend..first_path.."aligned_"..id..".tif "
+		else
+			images_to_blend = images_to_blend..'"'..first_path.."aligned_"..id..".tif"..'" '
+		end
+	end
+	BlendStartCommand=build_execute_command(enfuse_path, blend_args, images_to_blend)
+	
+	dt.print_log("Blend Start Command: "..BlendStartCommand)
+	resultBlend=dt.control.execute(BlendStartCommand)
+	dt.print_log("Completed Blend")
+
+	return resultBlend, path_with_filename
+end
+local function show_status(storage, image, format, filename, --Store: Called on each exported image
+  number, total, high_quality, extra_data)
+     dt.print(_('export TIFF for image fusion ')..tostring(truncate(number)).." / "..tostring(truncate(total)))   
+end
+function create_image_fusion(storage, image_table, extra_data) --Finalize: Called once when all images are done exporting	
+--Create Images String--
+	images_to_align = ""
+	images_to_blend = ""
+	counted_images=0
+	first_id = "999999999999999999999999"
+	last_id = "0"
+	for source_image,image_path in pairs(image_table) do
+		counted_images=counted_images+1
+		if dt.configuration.running_os == "windows" then
+			images_to_align = images_to_align..image_path..' '
```

Comment:
The windows version of align_image_stack will  not accept a filename with spaces in it, no matter how you quote it or escape it.  The linux version accepts filenames with spaces as long as they're wrapped in single quotes, so I think this is probably a bug in the windows implementation.

Your can scan the filename for spaces and then....

1. print an error to the screen about filenames not having spaces and exit

or 

2.  rename the exported files to something without spaces and modify the argument list to reflect that

or 

3.  leave it as is and report a bug to the hugin people and hope the fix comes out before somebody trips over the problem :-)

or 

4. leave it as is

Reply:
I'll probably go with #3 and then put a note in the readme / header warning windows users of this known issue.

Reply:
I checked this a little further and discovered it's not a bug with align_image_stack.  You can pass filenames with spaces, on windows, as long as they are surrounded with double quotes.  You can't, however, pass a double quoted executable path and a double quoted filename to dt.control.execute() because it fails with an error.  So, it appears that, until we find a work around, that you can't open filenames with spaces in external executables on windows.